chain/chain/src/chain.rs

```diff
@@ -3258,6 +3267,35 @@ impl<'a> ChainUpdate<'a> {
         Ok(())
     }
 
+    #[cfg(not(feature = "protocol_feature_block_header_v3"))]
+    fn verify_orphan_header_approvals(&mut self, _header: &BlockHeader) -> Result<(), Error> {
+        Ok(())
+    }
+
+    #[cfg(feature = "protocol_feature_block_header_v3")]
+    fn verify_orphan_header_approvals(&mut self, header: &BlockHeader) -> Result<(), Error> {
+        let prev_hash = header.prev_hash();
+        let prev_height = match header.prev_height() {
+            None => {
+                // this will accept orphans of V1 and V2
+                // TODO: reject header V1 and V2 after a certain height
```

Comment:
don't think we need to do this. We currently reject blocks that are older than tail right away, so after the network has been upgraded it wouldn't be long before the tail becomes V3

Reply:
Where do we reject them?
Seems like some checks are missing

Reply:
in `ClientActor::receive_block`, you might want to merge with master

Reply:
can't see the check, made `test_header_version_downgrade` with `should_panic`

Reply:
https://github.com/near/nearcore/blob/4a9ecaed34a9bce3ba7c469af48a78d9d02e8461/chain/client/src/client_actor.rs#L960

Reply:
We reject old heights, but don't reject old versions - a malicious node can produce V1 at any height, and it will be approved

Reply:
oh I misunderstood what you meant